### PR TITLE
Move xml stream join #201

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #201 I've moved xml  join stream implementation to test scope, extracted tests for this class and added one more test for xml join with buffered reader.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#204](https://github.com/artipie/rpm-adapter/issues/204) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/9b305e0d-6d77-4e08-82a2-5dd45577462a); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by g4s8** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/rpm-adapter/issues/204#issuecomment-634745619))

@g4s8 This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#204](https://github.com/artipie/rpm-adapter/issues/204) is not in scope
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Frpm-adapter%23204%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

File: src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            file, Paths.get(XmlStreamJoinTest.REPO, "primary.xml.example.second")
+        );
+        MatcherAssert.assertThat(
+            file,
+            CompareMatcher.isIdenticalTo(Paths.get(XmlStreamJoinTest.REPO, "primary.xml.example"))
+        );
+    }
+
+    @Test
+    void joinsWhenXmlIsNotProperlyFormatted(@TempDir final Path temp) throws IOException {
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java:63]: @olenagerasimova Something is strange about this test, both xmls used are properly formatted.

File: src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                "\n",
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+                "<parent>",
+                "<a>1</a><b>2</b><c>2</c><d>3</d>",
+                "</parent>"
+            ).getBytes()
+        );
+        MatcherAssert.assertThat(
+            target,
+            CompareMatcher.isIdenticalTo(expected)
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java:98]: @olenagerasimova Instead comparing if the two files are exactly the same, how about asserting if the generated file content xml is what we expect using xpath? I think that the test becomes more readable

File: src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java [hunk 3]
```diff
@@ -27,20 +27,11 @@
...
-import javax.xml.stream.XMLEventFactory;
-import javax.xml.stream.XMLEventReader;
-import javax.xml.stream.XMLEventWriter;
-import javax.xml.stream.XMLInputFactory;
-import javax.xml.stream.XMLOutputFactory;
-import javax.xml.stream.XMLStreamException;
-import javax.xml.stream.events.XMLEvent;
 
 /**
  * Joins two meta xml-files.
```
Comment (paulodamaso) [inline src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java:37]: @olenagerasimova Seems that we already have another class which joins two xml files, and do it better. So why keep this old implementation?

**Comment by g4s8** [comment]
@0crat assign @paulodamaso

**Comment by 0crat** [comment]
> @0crat assign @paulodamaso  ([here](https://github.com/artipie/rpm-adapter/issues/204#issuecomment-635138797))

@g4s8 This pull request [#204](https://github.com/artipie/rpm-adapter/issues/204) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/457d76ff-429c-4aa0-bc46-b7fd2739f02f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java (hunk 2 above)
Comment (olenagerasimova) [reply src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java:98]: @paulodamaso but less reliable: we will never know if resulting file contains some extra tags/data

File: src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java (hunk 3 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java:37]: @paulodamaso old implementation (with xml-streams) was moved to test scope and now is used in `XmlMetaJoinITCase` to compare resulting files.

**Comment by olenagerasimova** [comment]
@paulodamaso thanks, corrected and answered, please, take a look

File: src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java (hunk 2 above)
Comment (paulodamaso) [reply src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java:98]: @olenagerasimova Makes sense

File: src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java (hunk 1 above)
Comment (paulodamaso) [reply src/test/java/com/artipie/rpm/meta/XmlStreamJoinTest.java:63]: @olenagerasimova About this one, seems that `target` and `part` are correctly formatted, so the test name should be changed

File: src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java (hunk 3 above)
Comment (paulodamaso) [reply src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java:37]: @olenagerasimova Understood

**Comment by olenagerasimova** [comment]
@paulodamaso thanks again, I've corrected method name to be more precise

**Comment by paulodamaso** [comment]
@g4s8 good to merge

**Comment by olenagerasimova** [comment]
@g4s8 could you please take a look and merge this?

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**